internal/exec/exec.go

```diff
@@ -319,7 +327,7 @@ func (e *Exec) setupStdio() error {
 	// If the client requested a pseudo console then there's nothing we need to do pipe wise, as the process inherits the other end of the pty's
 	// pipes.
 	if e.cpty != nil && stdioRequested {
-		return errors.New("can't setup both stdio pipes and a pseudo console")
+		return nil
```

Comment:
So this was just crummy logic as anytime you'd do an exec you pretty much always are going to have stdout and stdin requested for, so this would fail beforehand.

Reply:
The new logic is to just skip creating the pipes (because we early return here). Then if you call the Stdin(), Stderr() or Stdout() methods we return either the inpipe or outpipe for the pty

Reply:
Rather than returning early here, can you set `stdioPipesOurSide` to hold the pipes from `cpty`?
That way we simplify some of the logic?

Reply:
I'd tried to reason with this also but the only bit that's weird is we definitely need to call `Close()` for the pty to close the handle for the console itself, which cleans up the pipes that we own inside the method also https://github.com/microsoft/hcsshim/blob/master/internal/conpty/conpty.go#L105. 

I'd tried to go with an approach where the Exec object isn't actually responsible for cleaning up some of the resources you pass into it (job object, pty and so on) and it's up to the creator to call `Close()` (or whatever the cleanup method is named for the resource, just so happens both of them are named Close haha), so it's solely responsible for things it made internally. If we assigned the ptys pipes to the array then we'd have to check if we asked for a pty and preferably skip the [`closeStdio`](https://github.com/microsoft/hcsshim/blob/master/internal/exec/exec.go#L381) method as the hope is the creator of the Exec object closes the pty themselves. Although we do ignore the errors from closing the stdio pipes in that method so nothing should actually freak out but *shrugs*

Reply:
Sorry, for more context double closing the pipe will cause an error such as 'close |0: file already closed', so if `closeStdio` ran before `Close` on the conpty object which it should , then on closing the conpty object you'd receive an error. So actually there is a small argument to keep it as it is

Reply:
Ah, I didnt track down the `close` portions and think that through.
sounds good to me then